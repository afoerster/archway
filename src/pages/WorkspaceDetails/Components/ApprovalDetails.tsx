import * as React from 'react';
import { Icon, Card } from 'antd';
import Label from './Label';
import { ApprovalItem } from '../../../types/Workspace';

interface ItemProps {
    approvalDate?: string;
    icon: string;
    children: any;
}

const Approval = ({ approvalDate, icon, children }: ItemProps) => (
  <div style={{ display: 'flex', flex: 1, flexDirection: 'column', justifyContent: 'center', alignItems: 'center' }}>
    <Icon
      type={approvalDate ? icon : 'dash'}
      theme={approvalDate ? 'twoTone' : 'outlined'}
      style={{ marginBottom: 5, fontSize: 28 }} />
    <div style={{ textTransform: 'uppercase', letterSpacing: 1 }}>{children}</div>
  </div>
);

interface Props {
    infra?: ApprovalItem;
    risk?: ApprovalItem;
}

const ApprovalDetails = ({ infra, risk }: Props) => (
  <Card
    style={{ display: 'flex', flex: 1 }}
    bodyStyle={{ display: 'flex', flexDirection: 'column', flex: 1 }}>
    <Label>approvals</Label>
    <div style={{ display: 'flex', flex: 1, alignItems: 'center' }}>
      <Approval approvalDate={risk && risk.approval_time} icon="safety-certificate">risk</Approval>
      <Approval approvalDate={infra && infra.approval_time} icon="dashboard">ops</Approval>
    </div>
  </Card>
);

export default ApprovalDetails;
