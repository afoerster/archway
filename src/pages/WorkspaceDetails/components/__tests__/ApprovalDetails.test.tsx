import * as React from 'react';
import { shallow } from 'enzyme';

import ApprovalDetails from '../ApprovalDetails';

describe('ApprovalDetails', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ApprovalDetails approveOperations={() => false} approveRisk={() => false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
