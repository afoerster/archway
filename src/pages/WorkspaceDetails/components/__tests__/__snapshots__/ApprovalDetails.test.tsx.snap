// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApprovalDetails renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ApprovalDetails
    approveOperations={false}
    approveRisk={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bodyStyle": Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
      },
      "children": Array [
        <Label>
          approvals
        </Label>,
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
            }
          }
        >
          <Approval
            approve={false}
          >
            risk
          </Approval>
          <Approval
            approve={false}
          >
            ops
          </Approval>
        </div>,
      ],
      "style": Object {
        "display": "flex",
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "approvals",
        },
        "ref": null,
        "rendered": "approvals",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Approval
              approve={false}
            >
              risk
            </Approval>,
            <Approval
              approve={false}
            >
              ops
            </Approval>,
          ],
          "style": Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "approvalDate": undefined,
              "approve": false,
              "children": "risk",
            },
            "ref": null,
            "rendered": "risk",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "approvalDate": undefined,
              "approve": false,
              "children": "ops",
            },
            "ref": null,
            "rendered": "ops",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bodyStyle": Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
        },
        "children": Array [
          <Label>
            approvals
          </Label>,
          <div
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flex": 1,
              }
            }
          >
            <Approval
              approve={false}
            >
              risk
            </Approval>
            <Approval
              approve={false}
            >
              ops
            </Approval>
          </div>,
        ],
        "style": Object {
          "display": "flex",
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "approvals",
          },
          "ref": null,
          "rendered": "approvals",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Approval
                approve={false}
              >
                risk
              </Approval>,
              <Approval
                approve={false}
              >
                ops
              </Approval>,
            ],
            "style": Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "approvalDate": undefined,
                "approve": false,
                "children": "risk",
              },
              "ref": null,
              "rendered": "risk",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "approvalDate": undefined,
                "approve": false,
                "children": "ops",
              },
              "ref": null,
              "rendered": "ops",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
